src/QueryM/Utxos.purs

```diff
@@ -157,7 +157,14 @@ getWalletCollateral = do
         addr <- liftAff $ (unwrap kw).address networkId
         utxos <- utxosAt addr <#> map unwrap >>> fromMaybe Map.empty
           >>= filterLockedUtxos
-        pure $ Array.singleton <$> (unwrap kw).selectCollateral utxos
+        pparams <- asks $ _.runtime >>> _.pparams <#> unwrap
+        let
+          coinsPerUtxoByte = pparams.coinsPerUtxoByte
+          maxCollateralInputs = fromMaybe 3 $ map UInt.toInt $
```

Comment:
Is it possible that the maximum is not set, i.e. any number is allowed?

If it is possible, let's not default to 3. If it is not possible, let's fail in case it's `Nothing`, because otherwise it would be very hard to debug.

Reply:
That default was taken from the original collateral return PR. I think it was done this way to limit the complexity of `selectCollateral`.

I think 3 is a reasonable number, given 5 inputs would guarantee collateral due to min ada restrictions. Collateral return reduces the impact of using larger value utxos anyway.

~I'll push the choice of 3 into the definition of `BalanceTx.Collateral.selectCollateral`, where there's already arbitrary choices for the sake of timely termination.~ This will be tackled in https://github.com/Plutonomicon/cardano-transaction-lib/issues/971 so perhaps we should do nothing here for now

Reply:
Thank you!